core/src/main/java/org/apache/calcite/rel/rules/PruneEmptyRules.java

```diff
@@ -540,4 +540,32 @@ public interface JoinRightEmptyRuleConfig extends PruneEmptyRule.Config {
       };
     }
   }
+
+
+  public static final RelOptRule EMPTY_TABLE =
```

Comment:
I think it would be a good idea to add the new rule to `org.apache.calcite.plan.RelOptRules#ABSTRACT_RULES` list along with the other pruning rules.